### PR TITLE
setup associations between models

### PR DESCRIPTION
**Pull request by sireclifford**
#### In this PR 
- Set up associations between models.
- Write SQL queries with Active Record.

> Methods
- A method that returns the 3 most recent posts for a given user.
- A method that updates the posts counter for a user.
- A method which returns the 5 most recent comments for a given post.
- A method that updates the comments counter for a post.
- A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,9 @@
 class User < ApplicationRecord
+  has_many :comments
+  has_many :posts
+  has_many :likes
```
Comment (KDlamini) [inline app/models/user.rb]: Kindly specify the foreign keys for posts, comments, and likes associations.

#### app/models/user.rb
```
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @sireclifford 🙋‍♀️,  

Great work so far! 💯 
There are some issues that you still need to work on to improve your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞 

✔️  Great job adding the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the Posts and Comments counters.
✔️  Good PR title and description
✔️  No linter errors

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (sireclifford) [reply app/models/user.rb]: thank you @KDlamini. I have rectify it.

**Review by rustam-tolipov — APPROVED** [review]
## Hello @sireclifford👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉 

No Linter errors 💯

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**